src/main/java/build/buildfarm/WorkerProfile.java

```diff
@@ -0,0 +1,354 @@
+// Copyright 2019 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buildfarm;
+
+import static java.lang.String.format;
+
+import build.buildfarm.common.DigestUtil;
+import build.buildfarm.common.redis.RedisClient;
+import build.buildfarm.instance.Instance;
+import build.buildfarm.instance.shard.JedisClusterFactory;
+import build.buildfarm.instance.stub.StubInstance;
+import build.buildfarm.v1test.OperationTimesBetweenStages;
+import build.buildfarm.v1test.RedisShardBackplaneConfig;
+import build.buildfarm.v1test.ShardWorker;
+import build.buildfarm.v1test.ShardWorkerConfig;
+import build.buildfarm.v1test.StageInformation;
+import build.buildfarm.v1test.WorkerProfileMessage;
+import build.buildfarm.worker.shard.WorkerOptions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Sets;
+import com.google.devtools.common.options.OptionsParser;
+import com.google.protobuf.Duration;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.TextFormat;
+import com.google.protobuf.util.Durations;
+import com.google.protobuf.util.JsonFormat;
+import io.grpc.ManagedChannel;
+import io.grpc.StatusRuntimeException;
+import io.grpc.netty.NegotiationType;
+import io.grpc.netty.NettyChannelBuilder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.naming.ConfigurationException;
+import redis.clients.jedis.JedisCluster;
+
+class WorkerProfile {
+  private static ManagedChannel createChannel(String target) {
+    NettyChannelBuilder builder =
+        NettyChannelBuilder.forTarget(target).negotiationType(NegotiationType.PLAINTEXT);
+    return builder.build();
+  }
+
+  /**
+   * Transform worker string from "ip-10-135-31-210.ec2:8981" to "10-135-31-210".
+   *
+   * @param worker
+   * @return
+   */
+  private static String workerStringTransformation(String worker) {
+    return worker.split("\\.")[0].substring("ip-".length()).replaceAll("-", ".") + ":8981";
```

Comment:
This is still pretty ugly, and speaks to a problem with DNS, hardcoded ports, and an implementation detail in doling out hostnames: I would honestly rather you have a DNS proxy/adapter in place that resolves these names than it be hardcoded in this tool.